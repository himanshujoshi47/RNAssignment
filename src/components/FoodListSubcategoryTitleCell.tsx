import React from 'react';
import {View, StyleSheet, Text} from 'react-native';
import R from '../res';

interface FoodListSubcategoryTitleCellProps {
  title: string;
}

const FoodListSubcategoryTitleCell: React.FC<FoodListSubcategoryTitleCellProps> = ({
  title,
}) => {
  return (
    <View>
      <Text testID={R.id.FOOD_LIST_ITEM_TITLE} style={localStyles.title}>
        {title}
      </Text>
      <View style={localStyles.separator} />
    </View>
  );
};

const localStyles = StyleSheet.create({
  container: {
    flex: 1,
  },
  title: {
    marginTop: R.dimensions.dimen_15,
    marginBottom: R.dimensions.dimen_16,
    fontSize: R.dimensions.dimen_18,
    color: R.colors.text.primary,
  },
  separator: {
    height: R.dimensions.dimen_1,
    backgroundColor: R.colors.background.silver,
  },
});

export default FoodListSubcategoryTitleCell;
