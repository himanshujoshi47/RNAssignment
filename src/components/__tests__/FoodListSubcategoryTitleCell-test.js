import React from 'react';
import renderer from 'react-test-renderer';
import R from '../../res';
import FoodListSubcategoryTitleCell from '../FoodListSubcategoryTitleCell';

jest.useFakeTimers();

describe('renders and have required elements', () => {
  let props = {
    title: 'Hey, Its me',
  };

  const foodListSubcategoryTitleCell = renderer.create(
    <FoodListSubcategoryTitleCell {...props} />,
  );
  const foodListSubcategoryTitleCellInst = foodListSubcategoryTitleCell.root;

  it('renders correctly', () => {
    expect(foodListSubcategoryTitleCell).toMatchSnapshot();
  });

  it('should have required elements', () => {
    expect(
      foodListSubcategoryTitleCellInst.findByProps({
        testID: R.id.FOOD_LIST_ITEM_TITLE,
      }),
    ).toBeTruthy();
  });
});
