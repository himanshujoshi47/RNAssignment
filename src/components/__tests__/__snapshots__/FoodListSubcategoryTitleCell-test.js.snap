// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and have required elements renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#222222",
        "fontSize": 18,
        "marginBottom": 16,
        "marginTop": 15,
      }
    }
    testID="food-list-item-title"
  >
    Hey, Its me
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#ECECEC",
        "height": 1,
      }
    }
  />
</View>
`;
